packages/deploy/src/utils.ts

```diff
@@ -3,6 +3,33 @@ export const uuidRegex = new RegExp(
   'i'
 );
 
+export const hyphenate = (str: any) => {
```

Comment:
This will do the job for now but unfortunately I think it's only a matter of time until it breaks.

It currently converts the spaces in names  to dashes, which matches Lightning behaviour. But what about other non-url safe characters? `@ ! ? # * %` etc, or emoji? Those will fail to sync correctly.

I really think the project payload should include a safe name (on everything, including edges and triggers) as well as a human readable label.

Reply:
something like this, @josephjclark ?

https://github.com/OpenFn/Lightning/issues/1053

Reply:
Yep, that should cover it